mautic/mautic
# Skip field import If value exists

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/mautic-documentation/pull/156
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | No
| Deprecations? | No

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
I got feature request from community to add option to import.
Skip field import If contact's field value already exists.
This is useful If we wanna import just data and not overwrite already existed data.

![image](https://user-images.githubusercontent.com/462477/56524548-e0cc1580-6549-11e9-9b52-571dad549a9c.png)


#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com/7460)
2. Try import data (email, customfield)
3. Then change customfield in contact profile
4. Then create new import and enable Skip field import If value exists
5. The customfield shouldn't be imported
6. Then remove value from customfield 
7. Create new import and enable Skip field import If value exists
8. The customfield should be imported
9. Same steps to test company import
